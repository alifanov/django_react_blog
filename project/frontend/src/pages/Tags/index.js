import { connect } from "react-redux";
import React, { Component } from "react";
import PropTypes from "prop-types";
import { getTags, addTag, delTag } from "../../actions/tags";
import { wsConnect } from "../../actions/websockets";
import { tagsCounterSelector, tagsSelector } from "../../selectors";
import { bindActionCreators } from "redux";

import Tag from "../../components/tag";
import TagForm from "../../components/forms/tag";

export class TagList extends Component {
  componentDidMount() {
    this.props.getTags();
    this.props.wsConnect();
  }

  render() {
    const { tags, counter } = this.props;
    return (
      <div>
        <TagForm onSubmit={name => this.props.addTag(name)} />
        <div>
          <h4>Tags: {counter}</h4>
        </div>
        {tags
          .map((p, i) => <Tag key={i} onDelete={this.props.delTag} {...p} />)
          .reduce((prev, curr) => [prev, " ", curr], null)}
      </div>
    );
  }
}

TagList.propTypes = {
  tags: PropTypes.array
};

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      getTags,
      addTag,
      delTag,
      wsConnect
    },
    dispatch
  );

const mapStateToProps = state => ({
  tags: tagsSelector(state),
  tagsCounter: tagsCounterSelector(state)
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(TagList);
