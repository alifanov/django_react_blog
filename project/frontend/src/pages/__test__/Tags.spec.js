import React from "react";
import Tags, { TagList } from "../tags";
import configureStore from "redux-mock-store";
import { shallow } from "../../enzyme";
import { MemoryRouter } from "react-router-dom";

import renderer from "react-test-renderer";

describe(">>>TAGS --- test smart component/page", () => {
  const initialState = { tagsReducer: { tags: ["1", "2", "3"] } };
  const mockStore = configureStore();
  let store, wrapper;

  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = shallow(<Tags store={store} />);
  });

  it("+++ render component", () => {
    expect(wrapper.length).toEqual(1);
  });

  it("+++ tags in prop", () => {
    expect(wrapper.props().tags).toEqual(initialState.tagsReducer.tags);
  });

  it("+++ check dummy", () => {
    const dummy = shallow(
      <TagList
        tags={[{ name: "1" }, { name: "2" }]}
        tagsCounter={1}
        getTags={() => null}
        wsConnect={() => null}
      />
    );
    expect(dummy.find("Tag").length).toEqual(2);
  });
});

describe(">>>Tags --- Snapshot", () => {
  it("+++capturing", () => {
    const renderedValue = renderer
      .create(
        <MemoryRouter>
          <TagList
            tags={[{ name: "1" }, { name: "2" }]}
            tagsCounter={1}
            getTags={() => null}
            wsConnect={() => null}
          />
        </MemoryRouter>
      )
      .toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
