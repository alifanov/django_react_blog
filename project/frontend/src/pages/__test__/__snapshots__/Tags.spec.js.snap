// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Tags --- Snapshot +++capturing 1`] = `
<div>
  <div
    className="jumbotron"
  >
    <h4>
      New tag
    </h4>
    <form
      className=""
    >
      <div
        className="position-relative form-group"
      >
        <label
          className=""
          htmlFor="name"
        >
          Name
        </label>
        <input
          className="form-control"
          id="name"
          name="name"
          onChange={[Function]}
          placeholder="Enter tag name"
          type="text"
          value=""
        />
      </div>
      <div
        className="d-flex justify-content-end"
      >
        <button
          aria-label={null}
          className="btn btn-secondary btn-sm"
          onClick={[Function]}
          type="button"
        >
          Add new post
        </button>
      </div>
    </form>
  </div>
  <div>
    <h4>
      Tags: 
    </h4>
  </div>
   
  <div
    className="btn-group"
    role="group"
  >
    <button
      aria-label={null}
      className="btn btn-outline-secondary"
      onClick={[Function]}
    >
      1
    </button>
    <button
      aria-label={null}
      className="btn btn-outline-secondary"
      onClick={[Function]}
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 352 512"
        width="1em"
      >
        <path
          d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        />
      </svg>
    </button>
  </div>
   
  <div
    className="btn-group"
    role="group"
  >
    <button
      aria-label={null}
      className="btn btn-outline-secondary"
      onClick={[Function]}
    >
      2
    </button>
    <button
      aria-label={null}
      className="btn btn-outline-secondary"
      onClick={[Function]}
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 352 512"
        width="1em"
      >
        <path
          d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        />
      </svg>
    </button>
  </div>
</div>
`;
